testing/file_test/README.md

```diff
@@ -88,6 +88,11 @@ Supported comment markers are:
     following CHECK lines without line information will immediately follow,
     between the CHECK with line information and the associated line.
 
+    STDERR checks are placed before the line they refer to if possible, and
+    otherwise as early in the file as possible. STDOUT check lines are placed
+    after the line they refer to if possible, and otherwise at the end of the
+    file.
```

Comment:
I feel like it would read a bit better integrating this with the previous paragraph. But I can't make a suggested edit that covers both, so just suggesting here for you to use or rework if useful, but also feel free to replace with your own version:

```
    When autoupdating, CHECKs will be inserted starting below AUTOUPDATE.
    For CHECKs without line information, STDERR will be placed immediately
    below AUTOUPDATE, and STDOUT will be at the end of the file.

    When a CHECK has line information, autoupdate will try to insert the
    CHECK and any following CHECK lines without line information together
    and next to that line if possible. STDERR CHECKs will go immediately
    above the line, and STDOUT CHECKs immediately after. The goal is to
    reflect that STDERR usually reflects validation of a line, and STDOUT
    the result of the line.
```

Reply:
Done and updated to match the new rules.